src/extractor/container.rs

```diff
@@ -370,18 +390,91 @@ impl<'a> Extractor<'a> {
-        Ok(links)
+        Ok((links, false))
+    }
+
+    /// Entry point to parse html for links (i.e. webscraping directory listings)
+    /// this function requests:
+    ///     http://localhost/<location>
+    pub(super) async fn parse_html(&self) -> Result<(HashSet<String>, bool)> {
+        log::trace!("enter: parse_html");
+
+        let mut links: HashSet<String> = HashSet::new();
+
+        // Response
+        let url = Url::parse(&self.url)?;
+        let response = self.make_extract_request(url.path()).await?;
+        let resp_url = response.url().as_str();
+        let body = response.text();
+        let html = Html::parse_document(body);
+
+        // Directory listing heuristic detection to not continue scanning
+        let title_selector = Selector::parse("title").unwrap();
+        for t in html.select(&title_selector) {
+            let title = t.inner_html().to_lowercase();
+            if title.contains("directory listing for /") || title.contains("index of /") {
+                log::debug!("Directory listing heuristic detection from \"{}\"", title);
+
+                self.get_links_by_attr(resp_url, &mut links, &html, "a", "href");
+                self.update_stats(links.len())?;
+
+                log::trace!("exit: parse_html -> {:?}", links);
+                return Ok((links, true));
+            }
+        }
+
+        // Get Links
+        self.get_links_by_attr(resp_url, &mut links, &html, "a", "href");
+        self.get_links_by_attr(resp_url, &mut links, &html, "img", "src");
+        self.get_links_by_attr(resp_url, &mut links, &html, "form", "action");
+        self.get_links_by_attr(resp_url, &mut links, &html, "script", "src");
+        self.get_links_by_attr(resp_url, &mut links, &html, "iframe", "src");
+        self.get_links_by_attr(resp_url, &mut links, &html, "div", "src");
+        self.get_links_by_attr(resp_url, &mut links, &html, "frame", "src");
+        self.get_links_by_attr(resp_url, &mut links, &html, "embed", "src");
+        self.get_links_by_attr(resp_url, &mut links, &html, "script", "src");
+
+        self.update_stats(links.len())?;
+
+        log::trace!("exit: parse_html -> {:?}", links);
+        Ok((links, false))
+    }
+
+    /// simple helper to get html links by tag/attribute and add it to the `links` HashSet
+    fn get_links_by_attr(
+        &self,
+        resp_url: &str,
+        links: &mut HashSet<String>,
+        html: &Html,
+        html_tag: &str,
+        html_attr: &str,
+    ) {
+        log::trace!("enter: get_links_by_attr");
+
+        let selector = Selector::parse(html_tag).unwrap();
+        let tags = html
+            .select(&selector)
+            .filter(|a| a.value().attrs().any(|attr| attr.0 == html_attr));
+        for t in tags {
+            if let Some(link) = t.value().attr(html_attr) {
+                log::debug!("Parsed link \"{}\" from {}", link, resp_url);
+                let mut new_url = Url::parse(&self.url).unwrap();
```

Comment:
unwrap here leads to runtime problems `thread 'tokio-runtime-worker' panicked at 'called `Result::unwrap()` on an `Err` value: RelativeUrlWithoutBase', src/extractor/container.rs:461:57`